### PR TITLE
make two heros from one data file

### PR DESCRIPTION
**Pull request by andrewpmiller**
## What this PR purports to do:
- move hero configuration to a (single) YAML data file
- generate two hero includes — one for all users and one for authenticated users

### Assumptions:
- LiveCoding toggle will overide settings for both hero includes
- The `current_promo` is the default for both hero includes
- The logic to include public/authenticated based on the logged-in state will be handled by the homepage template

### Test scenarios:

Public hero: Collections
Authenticated hero: Blue Beanie Day 2020

```yml
current_promo: hero-collections.html
auth_promo: hero-bbd-2020.html
live_coding: false
```
---

Public hero: Collections
Authenticated hero: Collections

```yml
current_promo: hero-collections.html
auth_promo: 
live_coding: false
```
---

Public hero: Live Coding
Authenticated hero: Live Coding

```yml
current_promo: hero-collections.html
auth_promo: 
live_coding: true
```
---

**Public Hero:** `/static/home/hero/`
https://deploy-preview-584--thegymcms.netlify.app/static/home/hero/

**Authenticated Hero:** `/static/home/hero-auth/`
https://deploy-preview-584--thegymcms.netlify.app/static/home/hero-auth/

### CODE REVIEW

File: _data/hero.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+current_promo: hero-collections.html
+auth_promo: hero-bbd-2020.html
+live_coding: false
```
Comment (rediris) [inline _data/hero.yml:3]: Could we change this to something more agnostic, something like `live_stream`, `livestream`, or something similar?

Comment (andrewpmiller) [reply _data/hero.yml:3]: Definitely. I'm thinking something more general... like: `roman_live_streaming_his_musical_performance` — cool?

Comment (rediris) [reply _data/hero.yml:3]: Perfeito.

Comment (andrewpmiller) [reply _data/hero.yml:3]: Actually, @rediris there's a (potentially moot) reason to keep `live_coding` as `live_coding` — as such. The underlying logic sets both public and authenticated hero to display the Twitch embed used for Live Coding sessions... based on the premise that the person who was doing the live coding would want a simple switch to toggle it on/off (while juggling other tasks).

The basis of this PR is the idea that we'd stream something for authenticated users only... so we wouldn't use the override that `live_coding` affords us. If/when we pilot this, we'd just make a `livestream` embed that would contain the appropriate player.

I'm not sure the _livecoding needs to be a quick switch_ premise is still valid though... but if we make this more generalized we need to consider how to support the myriad number of player embeds we'd want to support (and how).

Comment (andrewpmiller) [reply _data/hero.yml:3]: @jgagne — any thoughts on this ☝️ ?

Comment (jgagne) [reply _data/hero.yml:3]: I'm in favor of leaving the option for `live_coding` as is for now. The naming and option could easily be updated or removed depending on future use, but having the functionality to display unique heroes based on authentication states in a single file is a beautiful thing (and start).

Comment (rediris) [reply _data/hero.yml:3]: 👍 👍 👍 👍 👍

Comment (andrewpmiller) [reply _data/hero.yml:3]: @rediris and @jgagne — thanks! This is now un-drafted and ready for your review/approval/merge.

**Review by rediris — APPROVED** [review]

**Review by jgagne — APPROVED** [review]

**Pull request merged**